run_devstack_integration_tests.sh

```diff
@@ -14,34 +14,70 @@ paver update_assets lms --settings=test_static_optimized
 
 cp test_root/staticfiles/lms/webpack-stats.json test_root/staticfiles/webpack-stats.json
 
-
 cd /open-edx-plugins
 
-# Loop through each tar.gz file in the dist folder and install them using pip
-for file in "dist"/*.tar.gz; do
-    pip install "$file"
-done
+# Installing dev dependencies
+pip install poetry
+poetry install --no-interaction --only dev
+
+# Function to install or uninstall the plugin based on the subdirectory name and action
+manage_plugin() {
+    subdir=$1
+    action=$2
+    plugin_name=$(basename "$subdir" | sed 's/src\///' | sed 's/_/-/g')
+
+    tarball=$(ls dist | grep "$plugin_name" | head -n 1)
+
+    if [[ $action == "install" ]]; then
+        if [[ -n $tarball ]]; then
+            pip install "dist/$tarball"
+        else
+            echo "No matching tarball found for $plugin_name"
+            exit 1
+        fi
+    elif [[ $action == "uninstall" ]]; then
+        if [[ -n $plugin_name ]]; then
+            pip uninstall -y "$plugin_name"
+        else
+            echo "No plugin name found for $subdir"
+            exit 1
+        fi
+    else
+        echo "Invalid action: $action"
+        exit 1
+    fi
+}
 
 # Install codecov so we can upload code coverage results
 pip install codecov
 
 # output the packages which are installed for logging
 pip freeze
 
-set +e
-
-
 set +e
 for subdir in "src"/*; do
     if [ -d "$subdir" ]; then
         tests_directory="$subdir/tests"
 
         # Check if tests directory exists
         if [ -d "$tests_directory" ]; then
+            # Install the plugin from the subdirectory
+            manage_plugin "$subdir" "install"
+
             cp -r /edx/app/edxapp/edx-platform/test_root/ "/open-edx-plugins/$subdir/test_root"
             echo "==============Running $subdir test==================="
             cd "$subdir"
-            pytest . --cov .
+
+            # Check for the existence of settings/test.py
+            if [ -f "settings/test.py" ]; then
+                pytest_command="pytest . --cov . --ds=settings.test"
+            else
+                pytest_command="pytest . --cov ."
+            fi
+
+            # Run the pytest command
+            $pytest_command
+
```

Comment:
I think we can make this one simpler, Without `manage_plugin` function because we don't need to uninstall the package after running the tests. You can let me know if there is a reason we are uninstalling them.

Otherwise what I think we can do in this loop is:

1. Install the plugin, Add the log about which plugin is being tested
2. Run tests
3. Do other things, like coverage etc

Reply:
In the current scenario, yes we don't need to uninstall the plugin after tests because the tests of openedx-companion-auth run at the end after sysadmin and rapid response plugin. But in future if we add a plugin whose tests run after openedx-companion-auth's tests and the plugin requires login to run the test then the openedx-companion-auth will affect that.

Reply:
In a scenario you mentioned above, A plugin that needs `openedx-companion-auth` would have it in it's dependencies. That means the companion package will automatically be installed as part of that plugin.